README.md

```diff
@@ -738,46 +738,103 @@ Or by running a ``curl`` command like
-instance.incrementX();
-instance.x; // 11
+The compilation also generates a `src/js/newtonraphsonwasm.wasm` file which will be loaded with the `createModule` function.
+
+The WebAssembly module must be loaded and initialized by calling the `createModule` function and waiting for the promise to resolve.
+
+```{.js #wasmpromise}
+// this Javascript snippet is later referred to as wasmpromise
+createModule().then((module) => {
+  <<wasmcalc>>
+});
+```
+
+The `module` variable contains the `NewtonRaphson` class we defined in the binding above.
+
+The root finder can be called with.
+
+```{.js #wasmcalc}
+// this Javascript snippet is later referred to as wasmcalc
+const epsilon = 0.001;
+const finder = new module.NewtonRaphson(epsilon);
+const guess = -20;
+const root = finder.find(guess);
+```
+
+Append the root answer to the html page using [document manipulation functions](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append).
+
+```{.js #wasmcalc}
+const answer = document.createElement('span');
+answer.id = 'answer';
+answer.append(root);
+document.body.append(answer);
+```
+
+To run the JavaScript in a web browser a HTML page is needed.
+To get the `createModule` function we will import the `newtonraphsonwasm.js` with a script tag.
+
+```{.html file=src/js/example.html}
+<!doctype html>
+<!-- this HTML page is stored as src/js/example.html -->
+<html>
+  <script type="text/javascript" src="newtonraphsonwasm.js"></script>
+  <script>
+    <<wasmpromise>>
+  </script>
+</html>
+```
+
+The web browser can only load the `newtonraphsonwasm.js` file when hosted by a web server.
+Python ships with a built-in web server, we will use it to host the all files of the repository.
+
+```{.awk #host-files}
+python3 -m http.server 8000
 ```
 
-### Executing long running methods in Javascript
+Visit [http://localhost:8000/src/js/example.html](http://localhost:8000/src/js/example.html) to see the root answer.
+The root finding answer was calculated using the C++ algorithm compiled to a WebAssembly module, executed by some JavaScript and rendered on a HTML page.
+
+### Executing long running methods in JavaScript
 
-Executing a long running C++ method will block the browser from running any other code like updating the user interface. This is not very nice for the user. To run the method in the background, [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) have been defined. A web worker runs in its own thread and can be interacted with from Javascript using messages.
+Executing a long running C++ method will block the browser from running any other code like updating the user interface. This is not very nice for the user. To run the method in the background, [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) have been defined. A web worker runs in its own thread and can be interacted with from JavaScript using messages.
```

Comment:
```suggestion
Executing a long running C++ method will block the browser from running any other code like updating the user interface. In order to avoid this, the method can be run in the background using [web workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers). A web worker runs in its own thread and can be interacted with from JavaScript using messages.
```